python/tests/test_ld_matrix.py

```diff
@@ -0,0 +1,834 @@
+import io
+from itertools import combinations_with_replacement, permutations
+from typing import Any, Callable, Generator, Self
+
+import numpy as np
+import pytest
+
+import tskit
+
+
+class BitSet:
```

Comment:
I've modeled this class after the BitSet that I'm imagining for #2834.

The only strange thing about the API is that the intersect method saves to a third length-1 array, while the rest of the methods modify the calling instance. I suppose `union` and `difference` could write out to an `out` array like `intersect`, but it would be clunky and slightly confusing to be passing the calling instance back into the method.

In any case, I'm not sure how much effort should be put into this API since technically only the two-locus code is using it for now.